takeValue on null emission throws TurbineAssertionError

On a nullable flow, when trying to `takeValue` for a null emission, Turbine throws a `TurbineAssertionError` with error message:

```
Expected item but found Item(null)
app.cash.turbine.TurbineAssertionError: Expected item but found Item(null)
```

Repro test:
```
@Test fun `test take null item`() {
  val turbine = Turbine<String?>()
  turbine.add("one")
  turbine.add(null)

  assertEquals("one", turbine.takeItem())
  // throws TurbineAssertionError
  assertEquals(null, turbine.takeItem())
}
```

--- a/src/commonMain/kotlin/app/cash/turbine/channel.kt
+++ b/src/commonMain/kotlin/app/cash/turbine/channel.kt
@@ -156,8 +156,10 @@ internal fun <T> ReceiveChannel<T>.takeEventUnsafe(): Event<T>? {
  * @throws AssertionError if the next event was completion or an error, or no event.
  */
 public fun <T> ReceiveChannel<T>.takeItem(name: String? = null): T {
-  val event = takeEvent()
-  return (event as? Event.Item)?.value ?: unexpectedEvent(name, event, "item")
+  return when (val event = takeEvent()) {
+    is Event.Item -> event.value
+    else -> unexpectedEvent(name, event, "item")
+  }
 }
 
 /**

--- a/src/commonTest/kotlin/app/cash/turbine/TurbineTest.kt
+++ b/src/commonTest/kotlin/app/cash/turbine/TurbineTest.kt
@@ -18,6 +18,7 @@ package app.cash.turbine
 import kotlin.test.Test
 import kotlin.test.assertEquals
 import kotlin.test.assertFailsWith
+import kotlin.test.assertNull
 import kotlin.test.assertSame
 import kotlinx.coroutines.CompletableDeferred
 import kotlinx.coroutines.CoroutineScope
@@ -160,9 +161,11 @@ class TurbineTest {
   @Test
   fun awaitItem() = runTest {
     val item = Any()
-    val channel = Turbine<Any>()
+    val channel = Turbine<Any?>()
     channel.add(item)
+    channel.add(null)
     assertSame(item, channel.awaitItem())
+    assertNull(channel.awaitItem())
   }
 
   @Test
@@ -245,9 +248,11 @@ class TurbineTest {
   @Test
   fun takeItem() = withTestScope {
     val item = Any()
-    val channel = Turbine<Any>()
+    val channel = Turbine<Any?>()
     channel.add(item)
+    channel.add(null)
     assertSame(item, channel.takeItem())
+    assertNull(channel.takeItem())
   }
 
   @Test

